sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -61,6 +62,9 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
       // When neither spark.sql.warehouse.dir nor hive.metastore.warehouse.dir is set,
       // we will set hive.metastore.warehouse.dir to the default value of spark.sql.warehouse.dir.
       val sparkWarehouseDir = sparkContext.conf.get(WAREHOUSE_PATH)
+      logInfo(s"Setting hive.metastore.warehouse.dir ($hiveWarehouseDir) to the value of " +
+        s"${WAREHOUSE_PATH.key} ('$sparkWarehouseDir').")
+      sparkContext.hadoopConfiguration.set("hive.metastore.warehouse.dir", sparkWarehouseDir)
       sparkContext.conf.set("hive.metastore.warehouse.dir", sparkWarehouseDir)
```

Comment:
yes, it is ok to remove it, and  it will be more clear that  `hive.metastore.warehouse.dir` and `WAREHOUSE_PATH ` are all at their right places.